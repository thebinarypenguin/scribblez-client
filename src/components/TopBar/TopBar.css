#TopBar {
  color: #f6f3f5;
  background-color: #5a89b9;
  position: sticky;
  top:0px;
}

#TopBar a,
#TopBar a:link,
#TopBar a:active,
#TopBar a:visited {
  color: #ffffff;
  text-decoration: none;
}

#TopBar a:hover {
  font-weight: bold;
}

#TopBar .Container {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0.5rem 1rem;

}

#TopBar ul {
  list-style-type: none;
}

#TopBar li {
  display:inline-block;
  margin-left: 1rem;
  font-size: 1.2rem;
}
