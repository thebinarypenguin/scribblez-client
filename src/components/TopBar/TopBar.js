import React from 'react';
import { Link } from 'react-router-dom';

import './TopBar.css';

class TopBar extends React.Component {

  static defaultProps = {
    links: [],
  };

  constructor(props) {

    super(props);

    this.generateLinks = this.generateLinks.bind(this);
  }

  generateLinks() {

    return this.props.links.map((link, i) => {

      return (
        <li key={i}>
          <Link to={link.target}>{link.name}</Link>
        </li>
      );
    });
  }

  render() {

    return (
      <div id="TopBar">
        <h1>Scribblez</h1>
        <ul>
          { this.generateLinks() }
        </ul>
    </div>
    );
  }
}

export default TopBar;
