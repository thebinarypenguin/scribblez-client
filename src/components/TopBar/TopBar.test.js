import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from "react-router-dom";
import renderer from 'react-test-renderer';

import TopBar from './TopBar';

it('renders without crashing', () => {

  const container = document.createElement('div');

  const subject = (
    <MemoryRouter>
      <TopBar />
    </MemoryRouter>
  );

  ReactDOM.render(subject, container);
  ReactDOM.unmountComponentAtNode(container);
});

it('renders the UI as expected', () => {

  const subject = (
    <MemoryRouter>
      <TopBar />
    </MemoryRouter>
  );

  const tree = renderer.create(subject).toJSON();
  expect(tree).toMatchSnapshot();
});
