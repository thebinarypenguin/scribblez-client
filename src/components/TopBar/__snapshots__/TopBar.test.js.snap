// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  id="TopBar"
>
  <div
    className="Container"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1>
        Scribblez
      </h1>
    </a>
    <ul />
  </div>
</div>
`;
